lighthouse-core/lib/dependency-graph/simulator/connection-pool.js

```diff
@@ -99,16 +105,29 @@ module.exports = class ConnectionPool {
     const origin = String(record.parsedURL.securityOrigin());
     /** @type {TcpConnection[]} */
     const connections = this._connectionsByOrigin.get(origin) || [];
-    const wasConnectionWarm = !!this._connectionReusedByRequestId.get(record.requestId);
-    const connection = connections.find(connection => {
-      const meetsWarmRequirement = wasConnectionWarm === connection.isWarm();
-      return meetsWarmRequirement && !this._connectionsInUse.has(connection);
+    // Sort connections by decreasing congestion window, i.e. warmest to coldest
+    const sortedConnections = connections.sort((a, b) => b.congestionWindow - a.congestionWindow);
+
+    const availableWarmConnection = sortedConnections.find(connection => {
+      return connection.isWarm() && !this._connectionsInUse.has(connection);
+    });
+    const availableColdConnection = sortedConnections.find(connection => {
+      return !connection.isWarm() && !this._connectionsInUse.has(connection);
     });
 
-    if (!connection) return null;
-    this._connectionsInUse.add(connection);
-    this._connectionsByRecord.set(record, connection);
-    return connection;
+    const needsColdConnection = !options.ignoreObserved &&
+        !this._connectionReusedByRequestId.get(record.requestId);
```

Comment:
meta point: it's really funky to be mixing this "connection reused" state that refers to observed network activity with what's "available" and "needed" for the simulation. I wish these were clearer